src/script/hydra-notify

```diff
@@ -114,10 +157,34 @@ while (1) {
                 buildFinished($build, @payload[1..$#payload]);
             } elsif ($channelName eq "step_finished") {
                 stepFinished(int($payload[0]), int($payload[1]));
+            } elsif ($channelName eq "eval_finished") {
+                my ($evalId) = @payload;
+                my $eval = $db->resultset('JobsetEvals')->find($evalId)
+                    or die "eval $evalId does not exist\n";
+                evalFinished($eval);
+            } elsif ($channelName eq "eval_job_failed") {
+                my ($project, $jobset, $job, $evalId) = @payload;
+                my $eval = $db->resultset('JobsetEvals')->find($evalId)
+                    or die "eval $evalId does not exist\n";
+                jobEvalFailed($eval, $project, $jobset, $job);
+            } elsif ($channelName eq "eval_failed") {
+                my ($_, $project, $jobset, $inputInfo) = @payload;
+                my $inputInfo = decode_json($inputInfo);
+                my @inputsWithName = ();
+                foreach my $k (keys %$inputInfo) {
+                    my ($inputObj) = @{$inputInfo->{$k}};
+                    $inputObj->{name} = $k;
+                    push @inputsWithName, $inputObj;
+                }
+                evalFailed($project, $jobset, \@inputsWithName);
             }
         };
         if ($@) {
             print STDERR "error processing message '$payload' on channel '$channelName': $@\n";
         }
+
```

Comment:
I don't think this is needed. My understanding is that it's used for the test, and I wrote in the test how one can do it differently, alleviating the need for `once`.